doc/reference/std/net/address.md

```diff
@@ -2,160 +2,196 @@
 
 Internet addresses as host-port pairs.
 
+The canonical representation of Internet addresses throughout the
+Gerbil standard library is that of a pair, with a host in the car and
+the port at the cdr.  The host can be a string or an IP address, which
+is a u8vector of the right size (4 for IPv4 and 16 for IPv6).  A
+_normalized_ address can only have an IP address as the host.
+
+The `:std/net/address` module provides utilities for Internet address
+processing.
+
 ::: tip usage
 (import :std/net/address)
 :::
 
-## ip4-address?
-::: tip usage
+## inet-address?
 ```
-(ip4-address? ...)
+(inet-address? obj)
 ```
-:::
 
+Returns true if `obj` is an Internet address.
 
-Please document me!
+## inet-address-string?
+```
+(inet-address-string? obj)
+```
 
-## ip4-address
-::: tip usage
+Returns true if `obj` is a string representation of Internet address,
+with the canonical form `"host:port"`.
+
+## inet-address
 ```
-(ip4-address ...)
+(inet-address obj)
 ```
-:::
 
-Please document me!
+Normalizes the Internet address represented by `obj`, which can be an
+(unormalized) host-port pair or a string representing an Internet
+address.
 
-## ip4-address-string?
-::: tip usage
+Example:
 ```
-(ip4-address-string? ...)
+> (inet-address "127.0.0.1:8080")
+(#u8(127 0 0 1) . 8080)
 ```
-:::
 
-Please document me!
+## resolve-address
+```
+(resolve-address obj)
+```
 
-## ip4-address-&gt;string
-::: tip usage
+Resolves through DNS and normalizes the address represented by `obj`,
+which can be a string or a host-port pair.
+
+Example:
 ```
-(ip4-address->string ...)
+> (resolve-address "www.google.com:443")
+(#u8(172 217 18 4) . 443)
+```
+
+## resolved-address?
+```
+(resolved-address? obj)
 ```
-:::
 
-Please document me!
+Returns true if `obj` is a normalized Internet address.
 
-## string-&gt;ip4-address
-::: tip usage
+
+## inet-address->string
 ```
-(string->ip4-address ...)
+(inet-address->string addr)
 ```
-:::
 
-Please document me!
+Returns the string representation of an Ineternet address.
 
-## ip6-address?
-::: tip usage
+## string->inet-address
 ```
-(ip6-address? ...)
+(string->inet-address str)
 ```
-:::
 
-Please document me!
+Converts the string `str` representing an Internet address to its normalized form.
 
-## ip6-address
-::: tip usage
+
+## ip-address?
 ```
-(ip6-address ...)
+(ip-address? obj)
 ```
-:::
 
-Please document me!
+Returns true if `obj` is an IP address or a string representation of an IP address.
 
-## ip6-address-string?
-::: tip usage
+Example:
 ```
-(ip6-address-string? ...)
+> (ip-address? #u8(127 0 0 1))
+#t
+> (ip-address? "127.0.0.1")
+#t
+> (ip-address? "localhost")
+#f
+```
+
+## ip-address
+```
+(ip-address obj) -> ip-address
 ```
-:::
 
-Please document me!
+Normalizes the IP address represented by `obj`.
```

Comment:
Including DNS request, crypto certificate validation, etc.?

Reply:
yes, it uses the system resolver.